### PR TITLE
feat(coderd/healthcheck): improve detection of STUN issues

### LINKED ISSUE #12165: feat: support bundle: automatically diagnose STUN issues in network support bundle
Part of https://github.com/coder/coder/issues/12160

The support bundle includes a netcheck report.
We can potentially automatically diagnose STUN issues by checking the netcheck output:
- Is there at least one configured STUN server,
- Can we reach all configured STUN servers,
- Does the address detected by STUN vary by STUN region (Hard NAT)

### PR DESCRIPTION
**Pull request by johnstcn**
Fixes https://github.com/coder/coder/issues/12165

- Adds checks to `coderd/healthcheck/derphealth` for STUN issues. For now, I'm just checking that there is at least one healthy STUN server, and that we don't see variable port mapping.
- Support bundle command will now run a netcheck
- ~Support bundle command now dumps any errors or warnings from deployment health to stderr.~ (Edit: this can be a separate PR)

### CODE REVIEW

File: coderd/healthcheck/derphealth/derp.go [hunk 1]
```diff
@@ -32,6 +32,8 @@ const (
 	warningNodeUsesWebsocket = `Node uses WebSockets because the "Upgrade: DERP" header may be blocked on the load balancer.`
 	oneNodeUnhealthy         = "Region is operational, but performance might be degraded as one node is unhealthy."
 	missingNodeReport        = "Missing node health report, probably a developer error."
+	noSTUN                   = "No nodes are capable of STUN, direct connections may not be possible."
+	stunMapVaryDest          = "STUN detected variable mapping by destination IP, you may be behind a hard NAT."
```
Comment (mtojek) [inline coderd/healthcheck/derphealth/derp.go:36]: I have no idea what I'm reading... so do users? For example, `direct connections may not be possible` clearly describes what does not work. Should we rephrase it to make it user-friendly?

File: coderd/healthcheck/derphealth/derp.go [hunk 2]
```diff
@@ -107,9 +109,26 @@ func (r *Report) Run(ctx context.Context, opts *ReportOptions) {
...
+	var stunCapableNodes int
+	for _, region := range r.Regions {
+		for _, node := range region.NodeReports {
+			if node.STUN.CanSTUN {
+				stunCapableNodes++
+			}
+		}
+	}
+	if stunCapableNodes == 0 {
+		r.Severity = health.SeverityWarning
```
Comment (mtojek) [inline coderd/healthcheck/derphealth/derp.go:128]: Does it still warn if STUN handling is disabled?

File: support/support.go [hunk 3]
```diff
@@ -520,3 +520,62 @@ func sanitizeEnv(kvs map[string]string) {
...
+		for _, warn := range b.Network.Netcheck.Warnings {
+			msgs = append(msgs, "Client netcheck: "+warn.String())
+		}
+	}
+
+	if b.Deployment.HealthReport == nil {
+		msgs = append(msgs, "Deployment health check missing from bundle!")
+	} else {
+		if b.Deployment.HealthReport.AccessURL.Error != nil {
+			msgs = append(msgs, "Deployment health: "+*b.Deployment.HealthReport.AccessURL.Error)
```
Comment (mtojek) [inline support/support.go:543]: Wouldn't it be better to replace `Deployment health:` with a section name, for example `Access URL:`?

Otherwise, we will get a large block of 

`Deployment health: `
`Deployment health: `
`Deployment health: `
`Deployment health: `
`Deployment health: `

Same question for all sections.

File: support/support.go [hunk 4]
```diff
@@ -520,3 +520,62 @@ func sanitizeEnv(kvs map[string]string) {
...
+		}
+		for _, warn := range b.Network.Netcheck.Warnings {
+			msgs = append(msgs, "Client netcheck: "+warn.String())
+		}
+	}
+
+	if b.Deployment.HealthReport == nil {
+		msgs = append(msgs, "Deployment health check missing from bundle!")
+	} else {
+		if b.Deployment.HealthReport.AccessURL.Error != nil {
```
Comment (mtojek) [inline support/support.go:542]: Maybe we can improve the design a bit -
Right now the Bundle must be aware of HealthReport internals, so whenever somebody modifies the report, they need to adjust the bundle logic too. Maybe we should add a new method to `HealthReport`, `Summarize(b *Bundle)` and let the health report decide what should be shared or not?

**Review by mtojek — COMMENTED** [review]
High level/design review

File: support/support.go (hunk 4 above)
Comment (johnstcn) [reply support/support.go:542]: I might move this to a separate PR; It feels like the precursor to a `coder healthcheck` CLI command.

File: support/support.go (hunk 3 above)
Comment (johnstcn) [reply support/support.go]: Going to take this out for now and address in a separate PR; there's some refactoring we may need to do for this.

File: coderd/healthcheck/derphealth/derp.go (hunk 2 above)
Comment (johnstcn) [reply coderd/healthcheck/derphealth/derp.go:132]: Good point, this should now not alert if there are no STUN servers configured.

File: support/support.go (hunk 3 above)
Comment (johnstcn) [reply support/support.go]: https://github.com/coder/coder/pull/12963

File: support/support.go (hunk 4 above)
Comment (johnstcn) [reply support/support.go]: https://github.com/coder/coder/pull/12963

**Review by mtojek — APPROVED** [review]
